### DISCUSSION TITLE
Force the model to build a certain amount of capacity in a specified area and time period

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#49] [Q&A: Analysis]
Thank you for your attention and time!

I just want to know how a user can force a model to build a certain amount of technical capacity in a specified area, say gas-cc-ccs? For a general analysis, gas-cc-ccs are not competitive, but if I want to force gas-cc-ccs to build 500 MW capacity in 2035 in the p60, is this possible? What should I do?

Thanks!

**Comment by vcelloho** [comment]
There's a number of ways to do this. My recommendation would be to take advantage of the existing prescription constraint in the model.

eq_forceprescription
https://github.com/NREL/ReEDS_OpenAccess/blob/1a59d23ac6adfff787c3038566c4eaacb42beee2/c_supplymodel.gms#L452

For this to work you'll need in b_inputs to have the year 2035 added to force_pcat(pcat,t) this conditional determines whether or not the constraint is active, typically after the early model solve years this constraint becomes inactive.

You'll also need to add to the noncumulative_prescriptions parameter so that it includes the capacity that you wish to add. In this case it would look something like. 

noncumulative_prescriptions('gas-cc-ccs','p60','2035')=500;

You will want for these parameter changes to take effect after they are currently populated in b_inputs.gms.

If this will be more than a one off instance, I recommend linking this to csvs files that allow you to define the added capacity more flexibility without hard coding values to the model.

**Reply by wanyingw193** [reply]
Sounds fair! I will take a try. Many thanks!
